prov/efa/src/rdm/rxr_pkt_type_req.c

```diff
@@ -487,18 +487,17 @@ ssize_t rxr_pkt_init_rtm(struct rxr_pkt_entry *pkt_entry,
 	rtm_hdr->flags |= RXR_REQ_MSG;
 	rtm_hdr->msg_id = txe->msg_id;
 
-	data_size = MIN(txe->total_len - data_offset,
-			txe->ep->mtu_size - rxr_pkt_req_hdr_size_from_pkt_entry(pkt_entry));
-	if (data_size + data_offset < txe->total_len) {
-
-		if (txe->max_req_data_size > 0)
-			data_size = MIN(data_size, txe->max_req_data_size);
+	if (data_size == -1) {
+		data_size = MIN(txe->total_len - data_offset,
+				txe->ep->mtu_size - rxr_pkt_req_hdr_size_from_pkt_entry(pkt_entry));
 
-		if (efa_mr_is_cuda(txe->desc[0])) {
-			if (txe->ep->sendrecv_in_order_aligned_128_bytes)
-				data_size &= ~(EFA_RDM_IN_ORDER_ALIGNMENT - 1);
-			else
-				data_size &= ~(EFA_RDM_CUDA_MEMORY_ALIGNMENT -1);
+		if (data_size + data_offset < txe->total_len) {
+			if (efa_mr_is_cuda(txe->desc[0])) {
```

Comment:
it seems the check 

```
if (txe->max_req_data_size > 0)
			data_size = MIN(data_size, txe->max_req_data_size);
```
is dropped. Why?

Reply:
`max_req_data_size` was introduce to acheive data size evenness previous without change the init funciton's signature. Now that we pass `data_size` from function argument, it is not useful any more, and has been completed removed.